Assets/UserProfileManager.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+[Serializable]
+public class UserProfile
+{
+    public string profilename;
+    public string pin;
+    public string profilepicturepath;
+}
+
+[Serializable]
+public class ProfileListWrapper
+{
+    public List<UserProfile> profiles;
+}
+
+public class UserProfileManager
+{
+    private List<UserProfile> profiles = new List<UserProfile>();
+    private UserProfile selectedprofile = null;
+
+    private string savePath => Application.persistentDataPath + "/profiles.json";
+
+    public void saveProfiles()
+    {
+        string json = JsonUtility.ToJson(new ProfileListWrapper { profiles = this.profiles });
+        File.WriteAllText(savePath, json);
+    }
+
+    public void loadProfiles()
+    {
+        if (File.Exists(savePath))
```

Comment:
Instead of checking to see if the file exists, it would be better to wrap it around a try-catch block and then log the exception with `Debug.LogError`. That way if JsonUtility fails to parse the json, it'll report it to the log instead of crashing (and it will also handle `File.ReadAllText`'s exception as well)